src/arrayMethodJoin.js

```diff
@@ -5,7 +5,39 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    // const arr = [...this];
+    let res = '';
+    let separatorCopy = separator;
+
+    for (let i = 0; i < this.length; i++) {
+      // if (isNaN(this[i])) {
+      //   this[i] = '';
+      // }
+
+      if (this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (this[i] === null) {
+        this[i] = '';
+      }
```

Comment:
maybe you can combine this statements 
if (this[i] === undefined && (this[i] === null) {
this[i] = ''; 
}